test/javascripts/acceptance/chat-test.js

```diff
@@ -948,78 +954,150 @@ acceptance(
+        "Unfollow button is gone"
+      );
 
-    // Click unfollow!
-    await click(".chat-channel-unfollow");
-    assert.notOk(
-      settingsRow.querySelector(".chat-channel-expand-settings"),
-      "Expand notifications button is gone"
-    );
-    assert.notOk(
-      settingsRow.querySelector(".chat-channel-unfollow"),
-      "Unfollow button is gone"
-    );
+      assert.ok(
+        settingsRow.querySelector(".chat-channel-preview"),
+        "Preview channel button is present"
+      );
+      assert.ok(
+        settingsRow.querySelector(".chat-channel-follow"),
+        "Follow button is present"
+      );
+    });
 
-    assert.ok(
-      settingsRow.querySelector(".chat-channel-preview"),
-      "Preview channel button is present"
-    );
-    assert.ok(
-      settingsRow.querySelector(".chat-channel-follow"),
-      "Follow button is present"
-    );
-  });
-});
+    test("Create channel modal", async function (assert) {
+      await visit("/chat/channel/9/Site");
+      await click(".edit-channel-membership-btn");
+      assert.ok(exists(".channel-list-popup"));
+      assert.ok(exists(".channel-list-popup .popup-link.browse"));
+      assert.ok(exists(".channel-list-popup .popup-link.create"));
+      await click(".channel-list-popup .popup-link.browse");
+      assert.equal(currentURL(), "/chat/browse");
+      await visit("/chat/channel/9/Site");
+      await click(".edit-channel-membership-btn");
+      await click(".channel-list-popup .popup-link.create");
+      assert.ok(exists(".create-channel-modal-modal"));
+
+      assert.ok(query(".create-channel-modal-modal .btn.create").disabled);
+      let categories = selectKit(
+        ".create-channel-modal-modal .category-chooser"
+      );
+      await categories.expand();
+      await categories.selectRowByValue("6"); // Category 6 is "support"
+      assert.equal(
+        query(
+          ".create-channel-modal-modal .create-channel-name-input"
+        ).value.trim(),
+        "support"
+      );
+      assert.notOk(query(".create-channel-modal-modal .btn.create").disabled);
+
+      let types = selectKit(".create-channel-modal-modal .type-chooser");
+      await types.expand();
+      await types.selectRowByValue("topic");
+
+      await fillIn("#choose-topic-title", "This is a test");
+      const topicRow = query(".controls.existing-topic label");
+      await click(topicRow);
+      const topicTitle = topicRow
+        .querySelector(".topic-title")
+        .innerText.trim();
+      assert.equal(
+        query(
+          ".create-channel-modal-modal .create-channel-name-input"
+        ).value.trim(),
+        topicTitle
+      );
+      assert.notOk(query(".create-channel-modal-modal .btn.create").disabled);
+
+      await click(".create-channel-modal-modal .btn.create");
+      const done = assert.async();
+      next(async () => {
+        assert.equal(currentURL(), "/chat/channel/88/Something");
+        done();
+      });
```

Comment:
Weird that this is needed 🤔 in theory it should wait for the settled state

Reply:
It's because the transition happens in a `next` to make sure the chat service has marked the channel as followed... it's needed